swap/src/api/request.rs

```diff
@@ -0,0 +1,667 @@
+
+                let addresses = context
+                    .db
+                    .get_addresses(peerId)
+                    .await
+                    .with_context(|| "Could not get addressess")?;
+
+                json!({
+                    "peerId": peerId.to_base58(),
+                    "addresses": addresses
+                })
+            }
+            Method::SwapStartDate => {
+                let swap_id = self
+                    .params
+                    .swap_id
+                    .context("Parameter swap_id is missing")?;
+
+                let start_date = context.db.get_swap_start_date(swap_id).await?;
+
+                json!({
+                    "start_date": start_date,
+                })
+            }
+            Method::Config => {
+                let data_dir_display = context.config.data_dir.display();
+                tracing::info!(path=%data_dir_display, "Data directory");
+                tracing::info!(path=%format!("{}/logs", data_dir_display), "Log files directory");
+                tracing::info!(path=%format!("{}/sqlite", data_dir_display), "Sqlite file location");
+                tracing::info!(path=%format!("{}/seed.pem", data_dir_display), "Seed file location");
+                tracing::info!(path=%format!("{}/monero", data_dir_display), "Monero-wallet-rpc directory");
+                tracing::info!(path=%format!("{}/wallet", data_dir_display), "Internal bitcoin wallet directory");
+
+                json!({
+                    "result": []
+                })
+            }
+            Method::WithdrawBtc => {
+                let bitcoin_wallet = context
+                    .bitcoin_wallet
+                    .as_ref()
+                    .context("Could not get Bitcoin wallet")?;
+
+                let address = self
+                    .params
+                    .address
+                    .clone()
+                    .context("Parameter address is missing")?;
+
+                let amount = match self.params.amount {
+                    Some(amount) => amount,
+                    None => {
+                        bitcoin_wallet
+                            .max_giveable(address.script_pubkey().len())
+                            .await?
+                    }
+                };
+                let psbt = bitcoin_wallet
+                    .send_to_address(address, amount, None)
+                    .await?;
+                let signed_tx = bitcoin_wallet.sign_and_finalize(psbt).await?;
+
+                bitcoin_wallet
+                    .broadcast(signed_tx.clone(), "withdraw")
+                    .await?;
+
+                json!({
+                    "signed_tx": signed_tx,
+                    "amount": amount.to_sat(),
+                    "txid": signed_tx.txid(),
+                })
+            }
+            Method::StartDaemon => {
+                let addr2 = "127.0.0.1:1234".parse()?;
```

Comment:
Can we add an option to allow the user to set the daemon listen address manually?

Reply:
Yes of course. It's meant to be that way, I haven't changed from the default value back yet